code/parachain/frame/assets-transactor-router/src/lib.rs

```diff
@@ -0,0 +1,524 @@
+			asset_info: AssetInfo<T::Balance>,
+			amount: T::Balance,
+			governance_origin: <T::Lookup as StaticLookup>::Source,
+			dest: <T::Lookup as StaticLookup>::Source,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+
+			T::AssetsRegistry::register_asset(asset_id, None, asset_info)?;
+			let governance_origin = T::Lookup::lookup(governance_origin)?;
+			T::GovernanceRegistry::set(asset_id, SignedRawOrigin::Signed(governance_origin));
+			let dest = T::Lookup::lookup(dest)?;
+			<Self as fungibles::Mutate<T::AccountId>>::mint_into(asset_id, &dest, amount)?;
+			Ok(())
+		}
+
+		/// Mints `amount` of `asset_id` into the `dest` account.
+		#[pallet::weight(T::WeightInfo::mint_into())]
+		pub fn mint_into(
+			origin: OriginFor<T>,
+			asset_id: T::AssetId,
+			dest: <T::Lookup as StaticLookup>::Source,
+			amount: T::Balance,
+		) -> DispatchResult {
+			Pallet::<T>::ensure_admin_or_governance(origin, &asset_id)?;
+			let dest = T::Lookup::lookup(dest)?;
+			<Self as fungibles::Mutate<T::AccountId>>::mint_into(asset_id, &dest, amount)?;
+			Ok(())
+		}
+
+		/// Burns `amount` of `asset_id` into the `dest` account.
+		#[pallet::weight(T::WeightInfo::burn_from())]
+		pub fn burn_from(
+			origin: OriginFor<T>,
+			asset_id: T::AssetId,
+			dest: <T::Lookup as StaticLookup>::Source,
+			amount: T::Balance,
+		) -> DispatchResult {
+			Pallet::<T>::ensure_admin_or_governance(origin, &asset_id)?;
+			let dest = T::Lookup::lookup(dest)?;
+			<Self as fungibles::Mutate<T::AccountId>>::burn_from(asset_id, &dest, amount)?;
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		/// Returns `Ok(())` if origin is root or asset is signed by root or by origin
+		pub(crate) fn ensure_admin_or_governance(
+			origin: OriginFor<T>,
+			asset_id: &T::AssetId,
+		) -> Result<(), DispatchError> {
+			// TODO: that must be ensure_asset_origin(origin, asset_id))
+			if T::AdminOrigin::ensure_origin(origin.clone()).is_ok() {
+				return Ok(())
+			}
+
+			match origin.into() {
+				Ok(frame_system::RawOrigin::Signed(account)) => {
+					match T::GovernanceRegistry::get(asset_id) {
+						Ok(SignedRawOrigin::Root) => Ok(()), /* ISSUE: it says if */
+						// (call_origin.is_signed &&
+						// asst_owner.is_root) then allow
+						// mint/burn -> anybody can mint and
+						// burn PICA?
+						// TODO: https://app.clickup.com/t/37h4edu
+						Ok(SignedRawOrigin::Signed(acc)) if acc == account => Ok(()),
+						_ => Err(DispatchError::BadOrigin),
+					}
+				},
+				Ok(frame_system::RawOrigin::Root) => Ok(()),
+				_ => Err(DispatchError::BadOrigin), /* ISSUE: likely will not support collective
+													* origin which is reasonable to have for
+													* governance
+													https://app.clickup.com/t/37h4edu
+													*/
+			}
+		}
+	}
+
+	impl<T: Config> CreateAsset for Pallet<T> {
+		type LocalAssetId = T::AssetId;
+		type ForeignAssetId = T::AssetLocation;
+		type Balance = T::Balance;
+
+		fn create_local_asset(
+			protocol_id: [u8; 8],
```

Comment:
1. For IBC. It has asset incoming to be like `/portA/channelA/portB/channelB/42`. So it fit only `protocol_id`.  What would be nonce? 

2. Why we do not support assets without names/symbols/decimals. These are valid derived technical assets when protocol does not cares. Neither IBC nor XCM know these. If there is consideration to allow protocols (no govenrnae) to creae asset then asset ED/ratio/id should be possible without UI/FE metadata.

Reply:
What is relation of `protocol_id` to `AssetType`? 
Is it like /<assetType>/protocol_id/nonce?

Reply:
> Why we do not support assets without names/symbols/decimals.

This was done because the traits used by parity for getting asset metadata did not provide informative ways to fail. They must return something. The whole idea of always returning something felt incorrect at the time and still does now. An alternative would be returning `Default` values for both assets that don't exist and for ones that don't have metadata.

https://paritytech.github.io/substrate/master/frame_support/traits/tokens/fungibles/metadata/trait.Inspect.html
https://paritytech.github.io/substrate/master/frame_support/traits/tokens/fungibles/trait.InspectMetadata.html

Reply:
> What is relation of protocol_id to AssetType?

`protoco_id`, in most cases, should be the pallet ID. It does not hold bearing on the `AssetType`.

Reply:
Default is ok imho. Or #NOT-DEFINED like value.